dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -2193,6 +2235,416 @@ const char* DynamicData::typekind_to_string(TypeKind tk) const
+    os << my_double;
+    type_string += " = " + os.str() + "\n";
+    break;
+  }
+  case TK_FLOAT128: {
+    ACE_CDR::LongDouble my_longdouble;
+    if (dd.get_float128_value(my_longdouble, 0) != DDS::RETCODE_OK) {
+      if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+        ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: failed to get_float128_value\n"));
+      }
+      return false;
+    }
+    std::stringstream os;
+    os << my_longdouble;
+    type_string += " = " + os.str() + "\n";
+    break;
+  }
+  case TK_STRING8: {
+    ACE_CDR::Char* my_string = 0;
+    if (dd.get_string_value(my_string, 0) != DDS::RETCODE_OK) {
+      if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+        ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: failed to get_string_value\n"));
+      }
+      return false;
+    }
+    std::stringstream os;
+    string_helper(os, my_string);
+    type_string += DCPS::String(" = \"") + os.str() + "\"\n";
+    break;
+  }
+  case TK_STRING16: {
+    ACE_CDR::WChar* my_wstring = 0;
+    if (dd.get_wstring_value(my_wstring, 0) != DDS::RETCODE_OK) {
+      if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+        ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: failed to get_wstring_value\n"));
+      }
+      return false;
+    }
+    std::stringstream os;
+    string_helper(os, my_wstring);
+    type_string += " = L\"" + os.str() + "\"\n";
+    break;
+  }
+  case TK_BITMASK:
+    if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+      ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: Bitmask is an unsupported type in OpenDDS\n"));
+    }
+    break;
+  case TK_BITSET:
+    if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+      ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: Bitset is an unsupported type in OpenDDS\n"));
+    }
+    break;
+  case TK_SEQUENCE: {
+    DCPS::String temp_indent = indent;
+    indent += "  ";
+    ACE_CDR::ULong seq_length = dd.get_item_count();
+    type_string += "[" + DCPS::to_dds_string(seq_length) + "] =\n";
+    for (ACE_CDR::ULong i = 0; i < seq_length; ++i) {
+      type_string += indent + "[" + DCPS::to_dds_string(i) + "]";
+      if (dd.get_complex_value(temp_dd, i) != DDS::RETCODE_OK) {
+        if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+          ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: failed to get_complex_value\n"));
+        }
+        return false;
+      }
+      if (!print_dynamic_data(temp_dd, type_string, indent)) {
+        if (DCPS::log_level >= DCPS::LogLevel::Notice) {
+          ACE_ERROR((LM_NOTICE, "(%P|%t) NOTICE: print_dynamic_data: failed to read struct member\n"));
+        }
+        return false;
+      }
+    }
+    indent = temp_indent;
+    break;
+  }
+  case TK_ARRAY: {
+    DCPS::String temp_indent = indent;
+    indent += "  ";
+    LBound bound = dd.type()->get_descriptor().bound[0];
+    type_string += "[" + DCPS::to_dds_string(bound) + "] =\n";
```

Comment:
I'd prefer not to have braces at all in the output, but if we are going to have them in structs and unions I think we should be consistent and have them in arrays and sequences.

Reply:
I'll remove them from structs and unions.